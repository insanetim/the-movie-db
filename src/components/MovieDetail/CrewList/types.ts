import { ICrew } from 'src/interfaces/movie.interface'

export type CrewListProps = {
  crew: ICrew[]
}
