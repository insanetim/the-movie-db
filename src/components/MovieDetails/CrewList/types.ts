import { IMovieCredit } from 'src/interfaces/movie.interface'

export type CrewListProps = {
  crew: IMovieCredit[]
}

export type CrewListHookProps = {
  crew: IMovieCredit[]
}

export type CrewListHookReturn = {
  groupedCrew: {
    groupNames: string[]
  } & Partial<Record<string, IMovieCredit[]>>
}
