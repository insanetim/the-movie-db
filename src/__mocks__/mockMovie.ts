import type { IBackdrop, ICast, ICrew, IMovie, IMovieDetailExtended } from 'src/interfaces/movie.interface'

export const mockMovie: IMovie = {
  adult: false,
  backdrop_path: '/image',
  genre_ids: [1, 2, 3],
  id: 123,
  original_language: 'en',
  original_title: 'test/title',
  overview: 'test/overview',
  popularity: 123,
  poster_path: '/image',
  release_date: '01-01-1999',
  title: 'test/title',
  video: false,
  vote_average: 123,
  vote_count: 123
}

export const mockMovieDetail: IMovieDetailExtended = {
  accountStates: { favorite: false, id: 123, rated: false, watchlist: false },
  adult: false,
  backdrop_path: '/image',
  belongs_to_collection: null,
  budget: 1000000,
  credits: {
    cast: [
      {
        adult: true,
        cast_id: 123,
        character: 'test/character',
        credit_id: '123',
        gender: null,
        id: 123,
        known_for_department: '',
        name: 'test/name',
        order: 123,
        original_name: 'test/name',
        popularity: 123,
        profile_path: '/image'
      }
    ],
    crew: [
      {
        adult: true,
        credit_id: '123',
        department: 'test/department',
        gender: null,
        id: 123,
        job: 'test/job',
        known_for_department: '',
        name: 'test/name',
        original_name: 'test/name',
        popularity: 123,
        profile_path: '/image'
      }
    ],
    id: 123
  },
  genre_ids: [1, 2, 3],
  genres: [{ id: 1, name: 'test/genre' }],
  homepage: null,
  id: 123,
  images: [
    {
      aspect_ratio: 0.5625,
      file_path: '/image',
      height: 900,
      iso_639_1: null,
      vote_average: 123,
      vote_count: 123,
      width: 1600
    }
  ],
  imdb_id: null,
  original_language: 'en',
  original_title: 'test/title',
  overview: 'test/overview',
  popularity: 123,
  poster_path: '/image',
  production_companies: [{ id: 123, logo_path: null, name: 'test/company', origin_country: 'USA' }],
  production_countries: [{ iso_3166_1: 'en', name: 'USA' }],
  release_date: '01-01-1999',
  revenue: 3000000,
  runtime: 150,
  spoken_languages: [{ iso_639_1: 'en', name: 'EN' }],
  status: 'test/status',
  tagline: null,
  title: 'test/title',
  video: false,
  vote_average: 123,
  vote_count: 123
}

export const mockCast: ICast = {
  adult: true,
  cast_id: 123,
  character: 'test/character',
  credit_id: '123',
  gender: null,
  id: 123,
  known_for_department: '',
  name: 'test/name',
  order: 123,
  original_name: 'test/name',
  popularity: 123,
  profile_path: '/image'
}

export const mockCrew: ICrew = {
  adult: true,
  credit_id: '123',
  department: 'test/department',
  gender: null,
  id: 123,
  job: 'test/job',
  known_for_department: '',
  name: 'test/name',
  original_name: 'test/name',
  popularity: 123,
  profile_path: '/image'
}

export const mockImage: IBackdrop = {
  aspect_ratio: 0.5625,
  file_path: '/image',
  height: 900,
  iso_639_1: null,
  vote_average: 123,
  vote_count: 123,
  width: 1600
}
