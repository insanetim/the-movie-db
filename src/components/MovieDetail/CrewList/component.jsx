import PropTypes from 'prop-types'
import { Row, Col, Typography } from 'antd'

import CreditsItem from '../CreditsItem'

const CrewList = ({ crew }) => (
  <>
    <Col span={24}>
      <Typography.Title level={3}>Crew</Typography.Title>
    </Col>
    <Col span={24}>
      <Row gutter={[24, 16]}>
        {crew.map(item => (
          <Col
            key={item.credit_id}
            span={24}
            sm={12}
            md={8}
            lg={8}
            xl={6}
          >
            <CreditsItem
              profilePath={item.profile_path}
              title={item.name}
              description={item.job}
            />
          </Col>
        ))}
      </Row>
    </Col>
  </>
)

CrewList.propTypes = {
  crew: PropTypes.arrayOf(
    PropTypes.shape({
      credit_id: PropTypes.string,
      profile_path: PropTypes.string,
      name: PropTypes.string,
      job: PropTypes.string
    })
  ).isRequired
}

export default CrewList
