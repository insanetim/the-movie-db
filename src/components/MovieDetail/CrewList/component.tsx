import { Col, Row, Typography } from 'antd'

import CreditsItem from '../CreditsItem'
import { CrewListProps } from './types'

const CrewList: React.FC<CrewListProps> = ({ crew }) => (
  <>
    <Col span={24}>
      <Typography.Title level={3}>Crew</Typography.Title>
    </Col>
    <Col span={24}>
      <Row gutter={[24, 16]}>
        {crew.map(item => (
          <Col
            key={item.credit_id}
            lg={8}
            md={8}
            sm={12}
            span={24}
            xl={6}
          >
            <CreditsItem
              description={item.job}
              profilePath={item.profile_path}
              title={item.name}
            />
          </Col>
        ))}
      </Row>
    </Col>
  </>
)

export default CrewList
