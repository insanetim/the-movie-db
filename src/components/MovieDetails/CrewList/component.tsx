import { Col, Row, Typography } from 'antd'

import CreditsItem from '../CreditsItem'
import useContainer from './hook'
import { CrewListProps } from './types'

const CrewList: React.FC<CrewListProps> = ({ crew }) => {
  const { groupedCrew } = useContainer({ crew })

  return (
    <>
      <Col span={24}>
        <Typography.Title level={3}>
          Crew <span className='total-count'>{`(${crew.length})`}</span>
        </Typography.Title>
      </Col>
      {groupedCrew.groupNames.map(groupName => (
        <Col
          className='crew-group'
          key={groupName}
          span={24}
        >
          <Typography.Title level={4}>{groupName}</Typography.Title>
          <Row gutter={[24, 16]}>
            {groupedCrew[groupName]?.map(item => (
              <Col
                key={item.credit_id}
                lg={8}
                md={8}
                sm={12}
                span={24}
                xl={6}
              >
                <CreditsItem
                  description={item.job}
                  id={item.id}
                  profilePath={item.profile_path}
                  title={item.name}
                />
              </Col>
            ))}
          </Row>
        </Col>
      ))}
    </>
  )
}

export default CrewList
