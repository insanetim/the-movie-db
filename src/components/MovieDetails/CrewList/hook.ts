import { groupBy, sort } from 'ramda'
import { ICrew } from 'src/interfaces/movie.interface'

import { CrewListHookProps, CrewListHookReturn } from './types'

const useContainer = ({ crew }: CrewListHookProps): CrewListHookReturn => {
  const groups = groupBy((person: ICrew) => person.department, crew)
  const groupNames = sort(
    (a: string, b: string) => a.localeCompare(b),
    Object.keys(groups)
  )
  const groupedCrew = Object.assign(groups, { groupNames })

  return { groupedCrew }
}

export default useContainer
