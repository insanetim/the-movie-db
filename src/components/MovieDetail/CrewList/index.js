import CrewList from './component'

export default CrewList
