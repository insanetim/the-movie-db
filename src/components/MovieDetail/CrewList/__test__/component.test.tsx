import { mockCrew } from 'src/__mocks__/mockMovie'
import renderWithWrapper from 'src/utils/testHelpers/renderWithWrapper'

import CrewList from '../component'

describe('CrewList component', () => {
  it('should match snapshot', () => {
    const { asFragment } = renderWithWrapper(<CrewList crew={[mockCrew]} />)

    expect(asFragment()).toMatchSnapshot()
  })
})
