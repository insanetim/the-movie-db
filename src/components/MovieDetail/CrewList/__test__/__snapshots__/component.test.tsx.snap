// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CrewList component should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="ant-col ant-col-24 css-dev-only-do-not-override-6j9yrn"
  >
    <h3
      class="ant-typography css-dev-only-do-not-override-6j9yrn"
    >
      Crew
    </h3>
  </div>
  <div
    class="ant-col ant-col-24 css-dev-only-do-not-override-6j9yrn"
  >
    <div
      class="ant-row css-dev-only-do-not-override-6j9yrn"
      style="margin-left: -12px; margin-right: -12px; row-gap: 16px;"
    >
      <div
        class="ant-col ant-col-24 ant-col-sm-12 ant-col-md-8 ant-col-lg-8 ant-col-xl-6 css-dev-only-do-not-override-6j9yrn"
        style="padding-left: 12px; padding-right: 12px;"
      >
        <div
          class="ant-card ant-card-bordered css-dev-only-do-not-override-6j9yrn"
        >
          <div
            class="ant-card-cover"
          >
            <img
              alt="test/name"
              src="https://image.tmdb.org/t/p/w500/image"
            />
          </div>
          <div
            class="ant-card-body"
          >
            <div
              class="ant-card-meta"
            >
              <div
                class="ant-card-meta-detail"
              >
                <div
                  class="ant-card-meta-title"
                >
                  test/name
                </div>
                <div
                  class="ant-card-meta-description"
                >
                  test/job
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
